src/app/components/pickers/dateRange.tsx

```diff
@@ -1,67 +1,143 @@
-import React, {memo} from 'react';
-import {StyleSheet, View, ViewStyle} from 'react-native';
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
 
+import Modal from '../modal';
+import Typography from '../typography';
 import DatePicker from './datePicker';
-import Linear from '../seperator/linear';
+import Button from '../button';
 
 import strings from '../../constant/strings';
+import colors from '../../constant/colors';
+import fonts from '../../constant/fonts';
 
 type Props = {
-  style?: ViewStyle;
-  startDate: Date | undefined;
-  endDate: Date | undefined;
-  onChange: (date: Date | undefined, isStart: boolean) => void;
+  isVisible: boolean;
+  toggleModal: () => void;
+  onSubmit?: () => void;
+  startDate?: Date;
+  endDate?: Date;
+  onChangeStart: (date?: Date) => void;
+  onChangeEnd: (date?: Date) => void;
 };
 
-const DateRange = ({style, startDate, endDate, onChange}: Props) => {
+const DateRange = ({
+  isVisible,
+  startDate,
+  toggleModal,
+  onSubmit,
+  endDate,
+  onChangeEnd,
+  onChangeStart,
+}: Props) => {
+  const handleCancel = () => toggleModal();
```

Comment:
no need to create a new function we can directly pass `toggleModal` wherever passing `handleCancel`